Add notification banner for upcoming redesign

rainerdema opened the pull request:
## Summary

<!--
  Please include a summary of your changes, along with any useful context.

  You're encouraged to include screenshots in case of visual changes.

  If needed, you can reference other PRs or issues here with #ISSUE-NUMBER.
  You can use GitHub-specific syntax, e.g.



  However, if you do not have merge permissions on the repo, issues won't be auto-closed.
-->

Fixes #137 
 
This PR introduces a fixed header banner on the Solidus demo. The main aim is to inform users about our upcoming design overhaul.

https://github.com/solidusio/solidus-demo/assets/19948291/682bf03e-0ae2-4974-b0e4-be1da54dc391

<img width="1840" alt="Screenshot 2023-08-18 at 13 40 14" src="https://github.com/solidusio/solidus-demo/assets/19948291/5faedf85-55db-4b84-bd7b-0da3edb39d82">
<img width="976" alt="Screenshot 2023-08-18 at 13 40 19" src="https://github.com/solidusio/solidus-demo/assets/19948291/c7719f49-9eb3-43fb-9522-cb62115447ce">
<img width="612" alt="Screenshot 2023-08-18 at 13 40 23" src="https://github.com/solidusio/solidus-demo/assets/19948291/fa9911af-d439-49a8-a561-3595ddc92711">




## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

waiting-for-dev reviewed (approved):
Thanks, @rainerdema!

rainerdema commented:
@waiting-for-dev @aldesantis 
I've updated the text as requested and also changed the style of the banner (see the PR description).
Let me know what you think 👍

Here's the old one:

https://github.com/solidusio/solidus-demo/assets/19948291/32c3323f-3a41-4700-aaa3-b4764ef5c4b4

<img width="1840" alt="Screenshot 2023-08-17 at 17 00 55" src="https://github.com/solidusio/solidus-demo/assets/19948291/694783b2-4599-4398-b1a6-b4d9992e8c08">
<img width="612" alt="Screenshot 2023-08-17 at 17 00 51" src="https://github.com/solidusio/solidus-demo/assets/19948291/1a4abe00-d739-42f4-b4e3-15a1b7cff079">

waiting-for-dev reviewed (approved):
It looks great!
